code/modules/power/groundmap_geothermal.dm

```diff
@@ -589,6 +598,64 @@ GLOBAL_VAR_INIT(active_bluespace_generators, 0)
 	icon_state = "circ-on75-cold"
 
 
+/// Psychic mist -- Spawns on generator if it explodes; provides point gen as the generator would have
+/obj/machinery/mist_origin
+	name = ""
+	resistance_flags = RESIST_ALL|PROJECTILE_IMMUNE|DROPSHIP_IMMUNE
+	//Lists all visual mist effects
+	var/list/mist_list = list()
+	//What hive we should add psy points to
+	var/hivenumber
+
+/obj/machinery/mist_origin/Initialize(mapload, last_corrupted_hivenumber)
+	. = ..()
+	hivenumber = last_corrupted_hivenumber
+	for(var/turf/tile in filled_circle_turfs(src, 10))
+		var/obj/effect/psychic_mist/new_mist = new(tile)
+		mist_list += new_mist
+	start_processing()
+
+// Mimic generator point gen
+/obj/machinery/mist_origin/process()
+	if(!(length(GLOB.humans_by_zlevel["2"]) > 0.2 * length(GLOB.alive_human_list_faction[FACTION_TERRAGOV])))
+		return
+	if(GLOB.total_bluespace_generators) //Avoid dividing by 0
+		var/points_generated = GENERATOR_PSYCH_POINT_OUTPUT / GLOB.total_bluespace_generators
+		SSpoints.add_strategic_psy_points(hivenumber, points_generated)
+		SSpoints.add_tactical_psy_points(hivenumber, points_generated*0.25)
+
+/obj/machinery/mist_origin/Destroy()
+	GLOB.total_bluespace_generators-- //Remove bluespace generator from psy-gen equation, since we're no longer producing points
```

Comment:
Why not just have the gamemode itself generate the points based on the glob? This seems kinda fucky